T16 item availability child

dmunipi1 opened the pull request:
Hello reader,

This task is to make items seem unavailable on the child side.
When an item is unavailable it should have a red X icon over it and it should be blurred, unavailable items are not clickable.
I have separated the unavailable icon and "look" into a widget called item_unavailable.
Other helpful code you will find in child_boards (in pages) and clickable_images_grid (in widgets).
The images currently displayed are temporary and for testing purposes, that includes the category title and image. That needs to be resolved when the loading categories and items task is finished, I will be happy to help with that if needed.
If you have any questions or comments please let me know.

Thanks, 
Dion

dmunipi1 committed:
Add unavailable icon to all images

-Modity main file to run the child_board_interface file

dmunipi1 committed:
Add availability boolean to clickable_image model

-Add text under unavailable icon
-Separate text and icon from image in their own method
-Add if statement when loading images to check if available
-Change styling on text and icon

dmunipi1 committed:
Add keys to text and icon

dmunipi1 committed:
Merge main into branch
- Create item_unavailable widget
- Add comments to code
Merge branch 'master' of github.com:SEGAppProject/seg_coursework_app into T16-Item-Availability-Child

dmunipi1 committed:
Pull from main and fix conflicts

dmunipi1 committed:
Make images non clickable if unavailable

-Add comments to new methods
-Blur images if unavailable
-Adjust unavailable icon's size and alignment

dmunipi1 committed:
Lower opacity of blurred images

dmunipi1 committed:
Adjust colors of child board to match the app theme

dmunipi1 committed:
Update back button to take you to child timetables temporarily

dmunipi1 committed:
Add method to retrieve images and category image from database to child boards

dmunipi1 committed:
Merge with main branch and fix conflicts

dmunipi1 committed:
Fix back button

dmunipi1 committed:
Improve tests and restructure code

dmunipi1 committed:
Fix issue with tests

skellyhoe commented on lib/main.dart:
```diff
@@ -1,21 +1,28 @@
+  runApp(const MaterialApp(home: ChildInterface()));
+
+// Future main() async {
+//   WidgetsFlutterBinding.ensureInitialized();
```
Make sure to return main as it was.

skellyhoe commented on lib/pages/child_board/child_board.dart:
```diff
@@ -146,16 +163,20 @@ class _ChildBoards extends State<ChildBoards> with TickerProviderStateMixin {
+        hoverColor: Colors.transparent,
+        icon: const Icon(Icons.arrow_back_rounded),
+        onPressed: () =>
+            Navigator.of(context).pushReplacement(MaterialPageRoute(
```
Make sure when you pull from main to keep my changes (to pop the page instead of pushing a replacing page.)

skellyhoe commented on lib/pages/child_board/child_board.dart:
```diff
@@ -87,7 +104,7 @@ class _ChildBoards extends State<ChildBoards> with TickerProviderStateMixin {
         margin: const EdgeInsets.all(8.0),
         decoration: BoxDecoration(
-            color: Colors.red, borderRadius: BorderRadius.circular(8)),
+            color: Colors.teal, borderRadius: BorderRadius.circular(8)),
```
remove every hard coded color when you pull from main

skellyhoe commented on lib/pages/child_board/child_board.dart:
```diff
@@ -146,16 +163,20 @@ class _ChildBoards extends State<ChildBoards> with TickerProviderStateMixin {
+        alignment: Alignment.centerLeft,
+        padding: const EdgeInsets.all(8.0),
+        iconSize: 50,
+        splashColor: Colors.blue.shade900,
```
hard coded color

skellyhoe commented on test/screens/child/child_board_test.dart:
```diff
@@ -1,18 +1,35 @@
+  });
+
+  testWidgets(
+      'ChildBoards navigates back to the parent menu when the back button is pressed',
```
To do this test effectively

1. Start on customizable column and verify that customizable column is loaded.
2. Tap on a category. (Use pumpAndSettle() after this.)
3. Verify that the ChildBoards is loaded.
4. Tap the back button. (Use pumpAndSettle() after this.)
5. Verify that customizable column is loaded.

skellyhoe commented on lib/pages/child_board/child_board.dart:
```diff
@@ -146,16 +163,20 @@ class _ChildBoards extends State<ChildBoards> with TickerProviderStateMixin {
       decoration: BoxDecoration(
           border: Border.all(width: 3),
-          color: Colors.blue,
+          color: Color.fromARGB(255, 0, 76, 153),
```
hard coded color

skellyhoe reviewed (changes requested):
Good work! Clear and commented code!

I highlighted some issues that you should fix before merging into main.

However, for now, hard coded colors highlighted should be removed and the test highlighted should be refactored.

absirgue reviewed (commented):
Hello,

Thanks for the work!

In addition to Yousef's comments, can you try to make it easy for the person that will implement the fetching of the item data to transfer the data to this child_board page.

What I mean is that that person will have to get the data when the child "menu" is loaded and then when a category is clicked the data can just be based on to your class, doesn't have to be loaded again. In this regard, it would be nice to create a constructor for your class (or equivalent solution if you have a better idea) that takes the data of a category and its items and turn them into the now hardcoded categoryImage and images fields. 

For now this constructor can just set them to the hardcoded values you have here but at least the person that will implement this later task has an easy interface to work with.

Thanks!

dmunipi1 committed:
Pull main and fix conflicts

dmunipi1 committed:
Fix testing and child boards constructor

dmunipi1 commented:
Hello guys, 

Thanks for your comments and suggestions.
I have removed all hard coded colours.
Made the page to pop not push and left other changes that were on the main branch when I pulled.
I commented out the test Yousef mentioned for now because it is giving me a http error, I think its because the temporary images and their https (what I understood from the errors), and I will uncomment later into development when we have database queries sorted and when we remove the temporary images.
As per Anton's comment, that was already completed but it was commented out so maybe you didn't notice, I've moved the comments up to make it easier for the person implementing.

Thanks

dmunipi1 committed:
Fix test file for child boards
Add key to rows in CustomizableColumn

skellyhoe reviewed (approved):
Thank you! All good!

Ali-Alkhars reviewed (approved):
Nice and clean. Well done!

dmunipi1 committed:
Commit macOs changes

dmunipi1 committed:
Merge branch 'master' into T16-Item-Availability-Child
